pkg/controller/nodes/nodes_controller.go

```diff
@@ -288,3 +358,66 @@ func (r *ReconcileNodes) sendMarkedForTermination(oa *dynatracev1alpha1.OneAgent
 		},
 	})
 }
+
+func (r *ReconcileNodes) reconcileUnschedulableNode(node *corev1.Node, cache *Cache) error {
+	oneAgent, err := r.determineOneAgentForNode(node.Name)
+	if err != nil {
+		return err
+	}
+	if oneAgent == nil {
+		return nil
+	}
+
+	instance, hasNodeInstance := oneAgent.Status.Instances[node.Name]
+	if !hasNodeInstance {
```

Comment:
As determineOneAgentForNode() currently implemented, i.e., check for this Instances list and see if it's listed - this won't happen.